### PR TITLE
Development registration

### PR DESCRIPTION
**Pull request by maayarosama**
### Description

Adding registration and login views,  as well as adding the api's to them.

### Related Issues

#1

### CODE REVIEW

File: client/.env [hunk 1]
```diff
@@ -0,0 +1 @@
+VITE_API_ENDPOINT=http://localhost:3000/v1
```
Comment (xmonader) [inline client/.env]: why is this variable related to VITE? that's the api backend right? if so should be just `API_ENDPOINT` maybe

File: client/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,17 @@
+# stage 1
+FROM node:14.18.0-alpine as build
```
Comment (xmonader) [inline client/Dockerfile]: why not the latest LTS? 18.15 ?

File: client/src/components/Toast.vue [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+  import { createToast } from "mosha-vue-toastify";
+  export default {
+    setup() {
+      const toast = (title, color = "#217dbb") => {
+        createToast(title, {
+          position: "top-right",
+          hideProgressBar: true,
+          toastBackgroundColor: color,
+          timeout: 8000,
+          onClose: () => {
```
Comment (xmonader) [inline client/src/components/Toast.vue]: do we need to define this?  because we don't do any actions on Closing the toast

File: client/src/views/Forgetpassword.vue [hunk 4]
```diff
@@ -0,0 +1,117 @@
+<template>
+  <div class="div-wrapper">
+    <Toast ref="toast" />
+
+    <v-container>
+      <h5 class="text-h5 text-md-h4 text-center mt-10 mb-0 secondary">
+        Reset Password
+      </h5>
+      <div class="text-body-2 mb-10 text-center font-weight-light">The verification code will be sent to the mailbox.
```
Comment (xmonader) [inline client/src/views/Forgetpassword.vue]: to your mailbox instead of to the mailbox

File: client/src/views/Forgetpassword.vue [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+import axios from "axios";
+import Toast from "@/components/Toast.vue";
+
+export default {
+  components: {
+    Toast,
+  },
+
+  setup() {
+    var emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```
Comment (xmonader) [inline client/src/views/Forgetpassword.vue]: let's avoid vars completely either const or let

File: client/src/views/Forgetpassword.vue [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+    const toast = ref(null);
+
+    const router = useRouter();
+    const verify = ref(false);
+    const email = ref(null);
+    const loading = ref(false);
+    const isForgetpassword = ref(true);
+    const emailRules = ref([
+      value => !!value || 'Field is required',
```
Comment (xmonader) [inline client/src/views/Forgetpassword.vue:62]: what does the double negation do here for value?

File: client/src/views/Login.vue [hunk 7]
```diff
@@ -49,29 +50,75 @@
...
-        },
-        required(v) {
-            return !!v || 'Field is required'
-        },
+    components: {
+        Toast,
     },
-}
+    setup() {
+        var emailRegex = /^[^\s@]+@[^\s@]+\.[^\s@]+$/;
```
Comment (xmonader) [inline client/src/views/Login.vue]: let's use let or const instead of vars please

File: client/.env (hunk 1 above)
Comment (maayarosama) [reply client/.env]: when using vite, env vars should have VITE_API at the start. If it's not added env vars are set to undefined

**Review by samaradel — APPROVED** [review]

**Review by xmonader — APPROVED** [review]
